### ISSUE TITLE
"/tmp/tmp*" files left

### ISSUE THREAD

**Issue by JacekHoleczek** [ganga-devs/ganga#1402]
It seems that ganga-7.1.9 leaves many "/tmp/tmp*" files (one file per job).
This happens for any job that was resubmitted to "Slurm" (they just contain the standard submit output, like "Submitted batch job 14727079").
This doesn't happen for the "Local" backend, though.

You can easily test this behaviour, if you take [the "test_exit.py" script from one of my recent issues](https://github.com/ganga-devs/ganga/issues/1400) and change "exit 0" to "exit 1" inside (you may also need to change "Slurm" into your own backend, of course).

I've just checked that such files are now also always produced by the old ganga-6.0.44, even for jobs which were "completed" (I started to work on CentOS 7 with python 2.7.5; this problem did not exist on SL(C)6 with python 2.6).

**Comment by JacekHoleczek** [comment]
So, I think I have found the problem ...
In the "./GangaCore/Lib/Batch/Batch.py" file (ganga-7.1.9), in the "submit" method, there is:
```
                logger.info('could not match the output and extract the Batch queue name')

        # clean up the tmp file                                                                                                                                                                                                                                                
        if os.path.exists(soutfile):
            os.remove(soutfile)

        return rc == 0
```

This cleanup is completely missing in the "resubmit" and "kill" methods.
Please add it there (i.e. in both methods).

Here's a patch for the ganga-7.1.9 release (Ganga-7-1-9):

```diff
--- GangaCore/Lib/Batch/Batch.py.old	2019-01-13 10:14:00.000000000 +0000
+++ GangaCore/Lib/Batch/Batch.py	2019-01-14 09:22:20.000000000 +0000
@@ -298,6 +298,10 @@
             with open(soutfile) as sout_file:
                 logger.warning(sout_file.read())
 
+        # clean up the tmp file
+        if os.path.exists(soutfile):
+            os.remove(soutfile)
+
         return rc == 0
 
     def kill(self):
@@ -305,6 +309,9 @@
 
         with open(soutfile) as sout_file:
             sout = sout_file.read()
+        # clean up the tmp file
+        if os.path.exists(soutfile):
+            os.remove(soutfile)
         logger.debug('while killing job %s: rc = %d', self.getJobObject().getFQID('.'), rc)
         if rc == 0:
             return True
```

Here's a patch for Ganga-6-0-44 (GANGA 6.0.44 2015/3/27), Ganga-6-0-43 (GANGA 6.0.43 2015/1/29), Ganga-6-0-42-hotfix1 (GANGA 6.0.42-hotfix1 2015/1/14) and maybe older versions (but I don't have them installed so I cannot check it):

```diff
--- Ganga/Lib/Batch/Batch.py.old	2015-11-03 20:05:04.000000000 +0000
+++ Ganga/Lib/Batch/Batch.py	2019-01-14 09:56:24.000000000 +0000
@@ -202,6 +202,10 @@
             except IndexError:
                     logger.info('could not match the output and extract the Batch queue name')
                         
+        # clean up the tmp file
+        if os.path.exists(soutfile):
+            os.remove(soutfile)
+        
         return rc == 0
 
     def resubmit(self):
@@ -290,6 +294,10 @@
         else:
             logger.warning(file(soutfile).read())
                         
+        # clean up the tmp file
+        if os.path.exists(soutfile):
+            os.remove(soutfile)
+        
         return rc == 0
 
 
@@ -297,6 +305,9 @@
         rc,soutfile = self.command(self.config['kill_str'] % (self.id))
 
         sout = file(soutfile).read()
+        # clean up the tmp file
+        if os.path.exists(soutfile):
+            os.remove(soutfile)
         logger.debug('while killing job %s: rc = %d',self.getJobObject().getFQID('.'),rc)
         if rc == 0:
             return True
```

**Comment by JacekHoleczek** [comment]
The patch for the ganga-7.1.9 release (Ganga-7-1-9) is now a part of the https://github.com/ganga-devs/ganga/pull/1409 pull request.

**Comment by mesmith75** [comment]
Fixed by #1409
